docs/src/Provider.doc.js

```diff
@@ -19,65 +19,179 @@ card(
+      label: 'User Preference'
     }
   ];
   return (
-    <Provider colorScheme={scheme} id="docsExample">
-      <Box color="white" padding={2}>
+    <Provider colorScheme={scheme} id='docsExample'>
+      <Box color='white' padding={2}>
         <SelectList
-          id="scheme"
-          name="scheme"
+          id='scheme'
+          name='scheme'
           onChange={({ value }) => setScheme(value)}
           options={schemeOptions}
-          placeholder="Select color scheme"
-          label="Color scheme"
+          placeholder='Select color scheme'
+          label='Color scheme'
           value={scheme}
         />
         <Box padding={2}>
           <Text>Some content</Text>
         </Box>
-        <Button text="Example button" inline /> <Button color="red" text="Red Button" inline />
+        <Button text='Example button' inline /> <Button color='red' text='Red Button' inline />
       </Box>
     </Provider>
   );
 }`}
-  />,
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="OnNavigation"
+      description="This example illustrates the implemention of `onNavigation` context so that the link functionality of Gestalt components can be controlled externally. This example had 4 rellevant parts: a Provider, an `onNavigation` high-order function, consumer components (Link, Button, IconButton, TapArea), and `onNavigationOptions` props. The top Provider will pass to the consumer components the `onNavigation` function which will make accessible a final function that will be called during the `onClick` event handler. The `onNavigation` function can contain complex logic including React hooks to perform side effects. In this case, we are using `onNavigation` to disable the link, show an alert message, and open a different url in a new window. Finally, the `onNavigationOptions` props provides a flexible API. In this case, `navigationMode` allows to disable/enable the `onNavigation` logic. Other uses could be accessing `event.stopPropagation`."
+    >
+      <MainSection.Card
+        cardSize="lg"
+        defaultCode={`
+function OnNavigation() {
+  const [clientOnNavigationMode, setClientOnNavigationMode] = React.useState(true);
+
+  const onNavigation = ({ href, onNavigationOptions }) => {
+
+    const onNavigationClick = ({ event }) => {
+      event.nativeEvent.preventDefault();
+      // eslint-disable-next-line no-alert
+      alert('Disabled link. Opening help.pinterest.com instead.');
+      window.open('https://help.pinterest.com', '_blank');
+    }
+
+    return onNavigationOptions && onNavigationOptions.navigationMode === 'client'
+      ? onNavigationClick
+      : null;
+  }
+
+  const linkProps = {
+    href: 'https://pinterest.com',
+    onNavigationOptions: {
+      navigationMode: clientOnNavigationMode ? 'client' : 'server'
+    },
+    target: 'blank',
+  }
+
+  return (
+    <Provider onNavigation={onNavigation}>
+      <Flex direction='column' gap={2}>
```

Comment:
These should all use double quotes as well

Reply:
DONE